pyanaconda/payload/rpmostreepayload.py

```diff
@@ -368,15 +368,16 @@ def prepare_mount_targets(self, storage):
         # both mounts like /home (really /var/home) and %post scripts might
         # want to write to e.g. `/srv`, `/root`, `/usr/local`, etc. The
         # /var/lib/rpm symlink is also critical for having e.g. `rpm -qa` work
-        # in %post. We don't iterate *all* tmpfiles because we don't have the
-        # matching NSS configuration inside Anaconda, and we can't "chroot" to
-        # get it because that would require mounting the API filesystems in the
-        # target.
-        for varsubdir in ('home', 'roothome', 'lib/rpm', 'opt', 'srv',
-                          'usrlocal', 'mnt', 'media', 'spool', 'spool/mail'):
-            self._safe_exec_with_redirect("systemd-tmpfiles",
-                                          ["--create", "--boot", "--root=" + util.getSysroot(),
-                                           "--prefix=/var/" + varsubdir])
+        # in %post.
+        log.debug("copying over NSS files from sysroot")
+        nssfiles = ["passwd", "group"]
+        srcfiles = [util.getSysroot() + "/usr/lib/" + f for f in nssfiles]
+        cp_args = ["-a", "-bS", ".orig"] + srcfiles + ["/etc"]
+        self._safe_exec_with_redirect("cp", cp_args)
```

Comment:
First, please have this code silently do nothing if the files don't exist, so things continue to work after we [switch to sysusers](https://github.com/projectatomic/rpm-ostree/issues/49).

Second...I think you need to temporarily *concatenate* them, otherwise the `root` user won't be present, etc.

Reply:
Thanks for the review @cgwalters